ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -673,31 +692,44 @@ stdVersionDataNTC networkMagic = NodeToClientVersionData
     }
 
 stdRunDataDiffusion ::
-     DiffusionTracers
-  -> DiffusionArguments IO
-  -> DiffusionApplications
-       RemoteAddress
-       LocalAddress
-       NodeToNodeVersionData
-       NodeToClientVersionData
+     Diffusion.Tracers
+       RemoteAddress  NodeToNodeVersion
+       LocalAddress   NodeToClientVersion
+       IO
+  -> Diffusion.ExtraTracers p2p
+  -> Diffusion.Arguments
+       Socket      RemoteAddress
+       LocalSocket LocalAddress
+  -> Diffusion.ExtraArguments p2p IO
+  -> Diffusion.Applications
+       RemoteAddress  NodeToNodeVersion   NodeToNodeVersionData
+       LocalAddress   NodeToClientVersion NodeToClientVersionData
        IO
+  -> Diffusion.ExtraApplications p2p RemoteAddress IO
   -> IO ()
-stdRunDataDiffusion = runDataDiffusion
+stdRunDataDiffusion = Diffusion.run
 
 -- | Higher-level arguments that can determine the 'LowLevelRunNodeArgs' under
 -- some usual assumptions for realistic use cases such as in @cardano-node@.
 --
 -- See 'stdLowLevelRunNodeArgsIO'.
-data StdRunNodeArgs m blk = StdRunNodeArgs
+data StdRunNodeArgs m blk (p2p :: Diffusion.P2P) = StdRunNodeArgs
```

Comment:
Changing this data type changes the way that `cardano-node` invokes the Consensus layer. So: be advised they might need guidance about how to use these two new arguments etc.

Reply:
I have a corresponding pr here: https://github.com/input-output-hk/cardano-node/pull/2950

Reply:
btw, both are to `p2p-master` branch